### PR TITLE
Fix GatewayAction sensor not reset to default 

### PR DESCRIPTION
**Pull request by mikechouto**
- fixes issue #279 (Tested on `lumi.vibration.aq1`)
- base on PR #278 moved `reset_state` into `GatewayBinarySensor`  class .

Thank you @niceboygithub

### CODE REVIEW

**Comment by niceboygithub** [comment]
@EricCorleone  any option?

**Comment by EricCorleone** [comment]
As far as I know, the "unknown" state only appears when you reboot your HA and no event has been reported yet, and it should be the empty value `None` rather than the string "unknown".

**Comment by mikechouto** [comment]
I can change it to `none` instead of `unknown` but i'm sure that it not empty value in previous version of HA

**Comment by EricCorleone** [comment]
I'm not sure if you are talking about frontend display, because when the state is `None`, it will be displayed as "Unknown" on the frontend.

File: custom_components/aqara_gateway/binary_sensor.py [hunk 1]
```diff
@@ -714,9 +714,11 @@ def update(self, data: dict = None):
                 'entity_id': self.entity_id, 'click_type': self._state
             })
 
-            time.sleep(.1)
-
-            self._state = ''
+            # reset the state to empty after 0.1 second
+            if self.device['model'] in ('lumi.vibration.aq1'):
+                self.hass.loop.call_later(.1, self.reset_state('unknown'))
```
Comment (Necroneco) [inline custom_components/aqara_gateway/binary_sensor.py:719]: At least there is a syntax error here.

Comment (mikechouto) [reply custom_components/aqara_gateway/binary_sensor.py]: Hi, I run this code on my HA i don't seem to get any syntax error on my side can you kindly point it out ? or is it a coding style that i'm not matching ? Thanks

**Comment by mikechouto** [comment]
@niceboygithub @EricCorleone I've change the default value for the vibration sensor to `none` and repush the branch.

Thanks

**Comment by EricCorleone** [comment]
> @niceboygithub @EricCorleone I've change the default value for the vibration sensor to `none` and repush the branch.
> 
> Thanks

Sorry I forgot to mention that using the empty string `''` as the default value would be a better choice😂because it will be displayed completely blank on the frontend, which looks much better than "unknown". The "unknown" state only appears every time I reboot my HA as far as I know.
IMG_8888

**Comment by mikechouto** [comment]
> > @niceboygithub @EricCorleone I've change the default value for the vibration sensor to `none` and repush the branch.
> > Thanks
> 
> Sorry I forgot to mention that using the empty string `''` as the default value would be a better choice😂because it will be displayed completely blank on the frontend, which looks much better than "unknown" (which is `none` in the code). IMG_8888

Hi, sorry you might miss understood from my previous reply, when the state is `none` HA can correctly display it and it wouldn't show `unknown` please find the information in screenshot below. IMHO using a default state would be better than empty string as having a readable default state can bring more advantage when we write and maintain script/automation.

screenshot 2024-08-06 上午10 00 57

**Comment by EricCorleone** [comment]
> > > @niceboygithub @EricCorleone I've change the default value for the vibration sensor to `none` and repush the branch.
> > > Thanks
> > 
> > 
> > Sorry I forgot to mention that using the empty string `''` as the default value would be a better choice😂because it will be displayed completely blank on the frontend, which looks much better than "unknown" (which is `none` in the code). IMG_8888
> 
> Hi, sorry you might miss understood from my previous reply, when the state is `none` HA can correctly display it and it wouldn't show `unknown` please find the information in screenshot below. IMHO using a default state would be better than empty string as having a readable default state can bring more advantage when we write and maintain script/automation.
> 
> screenshot 2024-08-06 上午10 00 57

The `None` I meant is not a string, is the empty value in python without quotation marks and the first letter is capital. It will be rendered as "unknown" on the frontend according to my last try. But I guess the string "unknown" also works according to your screenshot. And the reason I prefer the empty string than "unknown" only because it looks better, but both are ok I guess (also I don't think I will use the empty/unknown state in automations).

**Comment by EricCorleone** [comment]
The `None` I meant is not a string, is the empty value in python without quotation marks and the first letter is capital. It will be rendered as "unknown" on the frontend according to my last try. But I guess the string "unknown" also works according to your screenshot. 

And the reason I prefer the empty string over "unknown" is simply because it looks better, but both are ok I guess (also I don't think I will use the empty/unknown state in automations).

**Comment by mikechouto** [comment]
Hey Eric yeah I've also tested the `None` in python which will be render as `unknown` on frontend. I think we can wait for @niceboygithub to see what he would like the default value to be for long term usage, then i can modify. Thanks again for the detail explain.

**Comment by Necroneco** [comment]
Both `""` and `"none"` are hardcoded. If I had to choose one, I'd choose `""`.
I searched the code history and found that it was always `""`. If you change this default, you might break other people's automation.

**Comment by mikechouto** [comment]
@Necroneco Thanks for pointing it out I've modify the branch to just moved `reset_state` into `GatewayBinarySensor` class and the function will always be `''` now.

**Review by Necroneco — APPROVED** [review]

**Comment by Necroneco** [comment]
But this is not much different from #280. Currently, only these two classes need this method.

**Comment by mikechouto** [comment]
@Necroneco I’m not sure if there are other devices that got effected as well as I don’d have a large variety to test against. And yeah merging both should be fine, I was just thinking to reduce duplicate code. ^_^
